src/main/io/osd.c

```diff
@@ -221,14 +222,19 @@ static void osdFormatPID(char * buff, const char * label, const pid8_t * pid)
     tfp_sprintf(buff, "%s %3d %3d %3d", label, pid->P, pid->I, pid->D);
 }
 
-static uint8_t osdGetHeadingIntoDiscreteDirections(int heading, int directions)
+static uint8_t osdGetHeadingIntoDiscreteDirections(int heading, unsigned directions)
```

Comment:
Yes, your formula gives exactly the same result so it seems good for me. The objective was to try to "let a code that is easy readable", this is the reason to divide the operation in different lines.
The result is the same, so I can change it. I don't know who decide this ;)

Reply:
@McGiverGim : It depends.. existing code is a bit longer and slower - unneeded first normalization, multiplication (FULL_CIRCLE * directions). Otherwise the code is equivalent.

BTW: `const unsigned segment = FULL_CIRCLE;` is IMO only complicating the code ...

Reply:
@ledvinap We can add maybe the constant to not let "magic numbers"? Or is innecesary for you?
```
direction = ((((heading + FULL_CIRCLE) * directions) + (FULL_CIRCLE/2)) / FULL_CIRCLE) % directions
```
@mikeller what do you think? This solution is right for you? I can change the code, but I need a final decision of the masters of betaflight :)

Reply:
@McGiverGim : Using FULL_CIRCLE is fine (personally I think that 360/180 is obvious when dealing with angles, but FULL_CIRCLE will work also for deci-degrees etc.)

You may also split equation into separate parts and comment them:
```
heading += FULL_CIRCLE;  // ensure positive value
int direction = (heading * directions + FULL_CIRCLE / 2) /  FULL_CIRCLE; // scale with rounding
direction %= directions; // normalize

Reply:
I was going for clarity over ultimate speed, but I think @ledvinap's last suggestion is clear and easy to understand.